server/middleware/users.coffee

```diff
@@ -57,3 +58,54 @@ module.exports =
     yield User.update({ _id: userID }, { $set: { "role": "student" } })
     user = yield User.findById req.user.id
     res.status(200).send(user.toObject({req: req}))
+
+  verifyEmailAddress: wrap (req, res, next) ->
+    user = yield User.findOne({ _id: mongoose.Types.ObjectId(req.params.userID) })
+    [timestamp, hash] = req.params.verificationCode.split(':')
+    if user and req.params.verificationCode is user.verificationCode(timestamp)
+      yield User.update({ _id: user.id }, { emailVerified: true })
+      res.status(200).send({ role: user.toObject().role })
+    else
+      next()
```

Comment:
I'm not sure I understand the use of `next` here. It seems like if the userID isn't found, that invalid input and should `throw new errors.UnprocessableEntity('No userID')` and if no user if found, `throw new errors.NotFound('User not found')`. What could `next` be passing control to?